src/trie/mod.rs

```diff
@@ -309,4 +334,25 @@ mod tests {
 
         assert_eq!(expected, actual);
     }
+
+    #[test]
+    fn on_collect_numbers_should_retrieve_numbers_from_decendent_nodes() {
```

Comment:
Look good. Maybe benchmarks for longer words?

Reply:
I don't think it will result with noticeable results.
1. I moved from root node to inner
2. I moved to node that branches off

These two should guarantee the correct results.

Though I think that benchmark for comparing the two recursive and loop is valid.

Reply:
Why I think we require a bigger dataset for tests:
 - If we have the word of length 20 and 20 nodes then we allocate to vector the same amount of time we create a new stack.
 - If we have a word of length 20 but 10_000 words -> nodes below the two letters prefix then we allocate to vector much more often then we create the new func call on and the stack... 
  
 Is that convincing you to go back to recursion? I was wrong I think. Can we prove (you prove) I was wrong and now I am correct or else with a larger dataset?

Reply:
Based on the Slack conversation I am convinced to make it not recursive.